### PR TITLE
Tasks/DES-2630: App Form styling

### PR DESCRIPTION
**Pull request by chandra-tacc**
## Overview: ##
This PR is first attempt at styling app form.

* Required tag logic changed in summary form. It only shows required if a field is required and does not have valid value. Definitely would like feedback on this.
* Used ant design theme where ever possible because some of the existing v3 code already moved in that direction.
* Some pending stuff:
** The Summary Panel on the right needs some padding and alignment adjustments (see env variables section). 
** A divider/bottom border is missing below Summary.
** Whether using styles attributes in some places ok, or should all of the design go with ant theme. 

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2630](https://tacc-main.atlassian.net/browse/DES-2630)

## Testing Steps: ##
1.  Get the PR locally
2. Start and go to https://designsafe.dev/rw/workspace/openfoam#!/

## UI Photos:
Shows that 
Screenshot 2024-05-16 at 9 18 58 AM

Screenshot 2024-05-16 at 9 19 22 AM
Screenshot 2024-05-16 at 9 19 28 AM

Screenshot 2024-05-16 at 9 19 33 AM
Screenshot 2024-05-16 at 9 19 45 AM
Screenshot 2024-05-16 at 9 19 51 AM

### CODE REVIEW

File: client/modules/workspace/src/AppsWizard/FormField.tsx [hunk 1]
```diff
@@ -60,37 +71,48 @@ export const FormField: React.FC<{
...
+          </div>
         )}
       </FormItem>
       {description && (
-        <small style={{ lineHeight: '20px' }}>{description}</small>
+        <small
+          style={{
+            display: 'block',
+            textAlign: 'left',
+            font: 'italic normal normal 10px/16px Helvetica Neue',
```
Comment (chandra-tacc) [inline client/modules/workspace/src/AppsWizard/FormField.tsx:107]: I took this font from xd. 10px might be too low for this.

File: client/modules/workspace/src/AppsWizard/Components.tsx
Comment (rstijerina) [inline client/modules/workspace/src/AppsWizard/Components.tsx:1]: This looks like a good candidate for a Common Component

File: client/modules/workspace/src/AppsWizard/FormField.tsx (hunk 1 above)
Comment (chandra-tacc) [reply client/modules/workspace/src/AppsWizard/FormField.tsx]: Upped the font size

File: client/modules/workspace/src/AppsWizard/Components.tsx
Comment (chandra-tacc) [reply client/modules/workspace/src/AppsWizard/Components.tsx:1]: Moved Primary and Secondary Button to common component.

**Review by rstijerina — APPROVED** [review]

**Review by wesleyboar — COMMENTED** [review]
This got merged. Understood. I know what the state of development is. I'll move these notes to JIRA ticket.

<details><summary><h4>1. Submit Button Color is Blue in Screenshot, but Green in Design</h4></summary>

Please prepare DS's button component to support different style buttons.[^1] I am not sure what colors to use for this and its states, because color from [design](https://xd.adobe.com/view/bab9a62d-3f2a-45f8-be0f-5680c474d9c1-efea/screen/ad7f3fda-3bc8-4aee-86ba-75d5da8299ee/) is #74B566, but white text on that is [not accessible](https://coolors.co/contrast-checker/ffffff-74b566).

@cs-jansen, please review [accessibility](https://coolors.co/contrast-checker/ffffff-74b566) of white on _that_ green. Shall I make a range of hues given all the states a button should have?

<details><summary>Consider using the DesignSafe green range of hues I've created.</summary>

The normal, dark, and x-dark are accessible with white text. The set is intentionally incomplete. [Full suite](https://huetone.ardov.me/?palette=N4IgdghgtgpiBcIAqBBAwmkAaEALArjAM4IDaoksCIEA9AIy0AOA9gE4AuEANtiAMYtu7EvFIgAxADMIUgCxSpfafRgAjKQBNlMAMyaA7FuWaArBsXKIugBwGY2nBIM2FMG8oBsABl0BOfn5lfXobPzVlACYbem9PT2VVb345IIBdAF8sCmg4RDpGfigSHEFhNlFxaV0pIzgnTSkYWXrJfghNRtaJCAh%2BIyUnG08IGEsnb19-P2VJyMj42e96el1dWcj6SN1vJe8IeZBM7PBc6jpIvjKRMkkbb3v75UeH3acpfilPGCCnGFNDPw3pIzGpTAZfncDAY5KZ1k5TPQ5LposEUfx6IM7q8nkMcbtjjkqPlaPCBEIbmJsS9nvidEZMcCJGpOg4Ik4-JFAvxTF56NYAso5HFNJ4PE5VGtdPRaTS8TTCadiSAAOZsCAATyuFIqtwkL1x1NesuN7wU5pMBkMhmUBm8drtQtczqi8zdJsN%2BrpisoeRAHFobAc2vKlSNnoNTMjyg%2BMDkmnZkg%2Bsj8Mr%2BmlSK2U-G2EF8yj8KLUcQ9Ue9WSJfoDGpg3GEAHcQ5SqtH5abw0ytFJvJoydIHDZPVJImoUQZs5o-AZ6JCJH4-J5uxBSyufWdEAG1TAYGAm7qqV65R2V041LIbJpZ4WvvxHJJYrUIDMnPp%2BDBOVExTZTFjD%2B2-08a7KgGajcIQe5hgBZZHlBMbeF84xtJECYyMEUjfvIUSmOYahpveMJPs%2Bx5toBFZKn6Fy0JoJTkqGeqTAxTKMZME5SH4mFOBAphxp0zypD8CQIjYED0OEwQuPQBiJhIyHmrOzFMcxRxkb65wMFRNHXPuVTDuhxgNNx7F3syDiKL%2BfiXuxci2poEAXnhEhrAYEBqOOTjRD4fhMpE3jxjCiTzGowqzMKGK8pkaQ4BwLBgMQtxQBAAAeAC03AAJYqrgHB8IluWpRlWU5Sl6WZdlODFQVZUgCVhU4GA7AJbwOC2WwADWOXJS17XlSlXVFb1EBtcpQA) (see `a/2/ds` row).

```
    "xx-light": "#cdf9f4",
    "x-light": "#a5e4dd",
    "light": "#84cec6",
    "normal": "#58a19b",
    "dark": "#37817b",
    "x-dark": "#2d4f4c",
```

</details>

[^1]: E.g. Core's [`primary`](https://tacc.utexas.edu/static/ui/components/detail/c-button--primary.html), [`secondary`](https://tacc.utexas.edu/static/ui/components/detail/c-button--secondary.html), [`tertiary`](https://tacc.utexas.edu/static/ui/components/detail/c-button--tertiary.html); Ant-D's [multiple types](https://ant.design/components/button#components-button-demo-multiple) and [danger type](https://ant.design/components/button#components-button-demo-danger).

</details>
<details><summary><h4>2. Form: Spacing & Borders & Font</h4></summary>

| design | screenshot |
| - | - |
| design | screenshot |

1. Font should be "Helvetica Neue".
2. There should be left and right padding for the item container.[^2]
3. The item containers are expected to have bottom gray border.
4. The spacing above and below headings does not match design.
5. There should be a line underneath "Summary".

[^2]: `ant-descriptions-item-container`?

</details>

**Comment by chandra-tacc** [comment]
Follow up task: [DES-2774](https://tacc-main.atlassian.net/browse/DES-2774) 
cc @wesleyboar

**Comment by chandra-tacc** [comment]
> This got merged. Understood. I know what the state of development is. I'll move these notes to JIRA ticket.
> 
> #### 1. Submit Button Color is Blue in Screenshot, but Green in Design
> Please prepare DS's button component to support different style buttons.[1](#user-content-fn-1-0e13d27471e2566c71553317c6818c9a) I am not sure what colors to use for this and its states, because color from [design](https://xd.adobe.com/view/bab9a62d-3f2a-45f8-be0f-5680c474d9c1-efea/screen/ad7f3fda-3bc8-4aee-86ba-75d5da8299ee/) is #74B566, but white text on that is [not accessible](https://coolors.co/contrast-checker/ffffff-74b566).
> 
> @cs-jansen, please review [accessibility](https://coolors.co/contrast-checker/ffffff-74b566) of white on _that_ green. Shall I make a range of hues given all the states a button should have?
> 
> Consider using the DesignSafe green range of hues I've created.
> The normal, dark, and x-dark are accessible with white text. The set is intentionally incomplete. [Full suite](https://huetone.ardov.me/?palette=N4IgdghgtgpiBcIAqBBAwmkAaEALArjAM4IDaoksCIEA9AIy0AOA9gE4AuEANtiAMYtu7EvFIgAxADMIUgCxSpfafRgAjKQBNlMAMyaA7FuWaArBsXKIugBwGY2nBIM2FMG8oBsABl0BOfn5lfXobPzVlACYbem9PT2VVb345IIBdAF8sCmg4RDpGfigSHEFhNlFxaV0pIzgnTSkYWXrJfghNRtaJCAh%2BIyUnG08IGEsnb19-P2VJyMj42e96el1dWcj6SN1vJe8IeZBM7PBc6jpIvjKRMkkbb3v75UeH3acpfilPGCCnGFNDPw3pIzGpTAZfncDAY5KZ1k5TPQ5LposEUfx6IM7q8nkMcbtjjkqPlaPCBEIbmJsS9nvidEZMcCJGpOg4Ik4-JFAvxTF56NYAso5HFNJ4PE5VGtdPRaTS8TTCadiSAAOZsCAATyuFIqtwkL1x1NesuN7wU5pMBkMhmUBm8drtQtczqi8zdJsN%2BrpisoeRAHFobAc2vKlSNnoNTMjyg%2BMDkmnZkg%2Bsj8Mr%2BmlSK2U-G2EF8yj8KLUcQ9Ue9WSJfoDGpg3GEAHcQ5SqtH5abw0ytFJvJoydIHDZPVJImoUQZs5o-AZ6JCJH4-J5uxBSyufWdEAG1TAYGAm7qqV65R2V041LIbJpZ4WvvxHJJYrUIDMnPp%2BDBOVExTZTFjD%2B2-08a7KgGajcIQe5hgBZZHlBMbeF84xtJECYyMEUjfvIUSmOYahpveMJPs%2Bx5toBFZKn6Fy0JoJTkqGeqTAxTKMZME5SH4mFOBAphxp0zypD8CQIjYED0OEwQuPQBiJhIyHmrOzFMcxRxkb65wMFRNHXPuVTDuhxgNNx7F3syDiKL%2BfiXuxci2poEAXnhEhrAYEBqOOTjRD4fhMpE3jxjCiTzGowqzMKGK8pkaQ4BwLBgMQtxQBAAAeAC03AAJYqrgHB8IluWpRlWU5Sl6WZdlODFQVZUgCVhU4GA7AJbwOC2WwADWOXJS17XlSlXVFb1EBtcpQA) (see `a/2/ds` row).
> 
> ```
>     "xx-light": "#cdf9f4",
>     "x-light": "#a5e4dd",
>     "light": "#84cec6",
>     "normal": "#58a19b",
>     "dark": "#37817b",
>     "x-dark": "#2d4f4c",
> ```
> 
> #### 2. Form: Spacing & Borders & Font
> design	screenshot
> design	screenshot
> 1. Font should be "Helvetica Neue".
> 2. There should be left and right padding for the item container.[2](#user-content-fn-2-0e13d27471e2566c71553317c6818c9a)
> 3. The item containers are expected to have bottom gray border.
> 4. The spacing above and below headings does not match design.
> 5. There should be a line underneath "Summary".
> 
> ## Footnotes
> 1. E.g. Core's [`primary`](https://tacc.utexas.edu/static/ui/components/detail/c-button--primary.html), [`secondary`](https://tacc.utexas.edu/static/ui/components/detail/c-button--secondary.html), [`tertiary`](https://tacc.utexas.edu/static/ui/components/detail/c-button--tertiary.html); Ant-D's [multiple types](https://ant.design/components/button#components-button-demo-multiple) and [danger type](https://ant.design/components/button#components-button-demo-danger). [↩](#user-content-fnref-1-0e13d27471e2566c71553317c6818c9a)
> 2. `ant-descriptions-item-container`? [↩](#user-content-fnref-2-0e13d27471e2566c71553317c6818c9a)

Thanks @wesleyboar  - will handle this in [DES-2774](https://tacc-main.atlassian.net/browse/DES-2774)

**Comment by wesleyboar** [comment]
@chandra-tacc You're faster than me 🙂. I put my piecemeal tickets ([DES-2775](https://tacc-main.atlassian.net/browse/DES-2775), [DES-2773](https://tacc-main.atlassian.net/browse/DES-2773)) as sub-tasks of yours ([DES-2774](https://tacc-main.atlassian.net/browse/DES-2774)). Feel free to close 'em out if someone will tackle those all at once.

**Pull request merged**